pkg/controller/pd_control.go

```diff
@@ -20,15 +20,19 @@ import (
 
 // getPDClientFromService gets the pd client from the TidbCluster
 func getPDClientFromService(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
-	if tc.IsHeterogeneous() {
+	if tc.HeterogeneousWithoutLocalPD() {
 		if len(tc.Spec.ClusterDomain) > 0 {
-			return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.Spec.ClusterDomain, tc.IsTLSClusterEnabled())
+			// FIXME: across k8s cluster without local pd
+			// if tls enable tc.Spec.Cluster.Name should be same as tc.Name? because it will query the secret using this name?
```

Comment:
I mean  in the across k8s cluster case(without local pd)

Reply:
I think there is a problem is that `tc.Spec.Cluster.Name` cannot use the same name and the same namespace with other clusters when considering TLS in the `clusterDomain` method.

Reply:
It means the cluster2 (the cluster joining the cluster1) need to store the TLS at the `tc.Spec.Cluster.NameSpace` and `tc.Spec.Cluster.Name`?

Reply:
> @july2993 Please also cover the TLS case, could you please also share the final arguments for each component?

update in PR description

Reply:
@july2993 
1. we need to request the users to create the cluster client secret in the Kubernetes cluster
2. Actually we can just use the cluster client secret of the original TC since they share  the same CA.